drivers/libusb0.c

```diff
@@ -396,44 +397,26 @@ static int libusb_open(usb_dev_handle **udevp,
 #endif
 
 			if (dev->descriptor.iManufacturer) {
-				retries = MAX_RETRY;
-				while (retries > 0) {
-					ret = usb_get_string_simple(udev, dev->descriptor.iManufacturer,
-						string, sizeof(string));
-					if (ret > 0) {
-						curDevice->Vendor = xstrdup(string);
-						break;
-					}
-					retries--;
-					upsdebugx(1, "%s get iManufacturer failed, retrying...", __func__);
+				ret = nut_usb_get_string(udev, dev->descriptor.iManufacturer,
+					string, sizeof(string));
+				if (ret > 0) {
+					curDevice->Vendor = xstrdup(string);
 				}
```

Comment:
It is technically redundant, because the general logging in nut_usb_get_string_descriptor() tells which descriptor was failing. But I agree it is easier to read something about "Manufacturer" instead of looking looking up what string index "1" was about. And I suppose they should have different log levels, with the one in nut_usb_get_string_descriptor() being only debug (since retries may have solved it) whereas the final failure can be critical for the device identification.

I am not familiar with the debug levels in use (or which direction they go). Can you please suggest suitable levels?

Reply:
The lower debug nunbers are more visible, but so far there are few formal criteria about what is exposed how. I guess the lines in `nut_usb_get_string_descriptor()` can be emitted with 2 or 3, so a hiccup getting those values (an improbable event per se?) can be seen with little noise from other events that would require a higher verbosity to troubleshoot.

Reply:
As mentioned in the other thread, if we are not sure if these retries will ever happen (?) then maybe we should make sure we'll always see them in logs, for collecting statistics on it.

Reply:
Does make sense too.